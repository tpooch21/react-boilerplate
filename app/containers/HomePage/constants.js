/**
 * HOME PAGE constants
 */

export const LOAD_STRINGS = 'stringDisplayer/HomePage/LOAD_STRINGS';
export const LOAD_STRINGS_SUCCESS =
  'stringDisplayer/HomePage/LOAD_STRINGS_SUCCESS';
