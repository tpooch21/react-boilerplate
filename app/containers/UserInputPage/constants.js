export const ENTER_STRING = 'stringDisplayer/UserInputPage/ENTER_STRING';
export const SUBMIT_STRING = 'stringDisplayer/UserInputPage/SUBMIT_STRING';
