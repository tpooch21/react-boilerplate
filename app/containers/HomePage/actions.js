import { LOAD_STRINGS, LOAD_STRINGS_SUCCESS } from './constants';

export const loadStrings = () => ({
  type: LOAD_STRINGS
});

export const loadStringsSuccess = () => ({
  type: LOAD_STRINGS_SUCCESS,
  strings,
});
