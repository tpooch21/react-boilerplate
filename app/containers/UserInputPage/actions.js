import { ENTER_STRING } from './constants';

export const enterString = input => ({
  type: ENTER_STRING,
  currentString: input,
});
