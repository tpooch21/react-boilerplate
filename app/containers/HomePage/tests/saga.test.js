import { put, takeLatest } from 'redux-saga/effects';

import getStringData, { getStrings } from '../saga';
import { loadStringsSuccess, loadStringsError } from '../actions';
import { LOAD_STRINGS } from '../constants';

describe('getStrings saga', () => {
  let getStringsGenerator;

  beforeEach(() => {
    getStringsGenerator = getStrings();

    // Test whether first yield (API call) is returning appropriate values from backend
    const callDescriptor = getStringsGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch loadStringsSuccess if data request is successful', () => {
    const strings = [{ id: 1, value: 'Test' }, { id: 1, value: 'Robot' }];

    // Since yield to call has already been called in beforeEach, 'next' now points to put
    const putDescriptor = getStringsGenerator.next(strings).value;
    expect(putDescriptor).toEqual(put(loadStringsSuccess(strings)));
  });

  it('should dispatch loadStringsError if data request fails', () => {
    const response = new Error('');

    const putDescriptor = getStringsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(loadStringsError()));
  });

  describe('getStringData saga', () => {
    const getStringDataGenerator = getStringData();

    it('should watch for LOAD_STRINGS actions and call getStrings when one is dispatched', () => {
      const takeLatestDescriptor = getStringDataGenerator.next().value;
      expect(takeLatestDescriptor).toEqual(
        takeLatest(LOAD_STRINGS, getStrings),
      );
    });
  });
});
