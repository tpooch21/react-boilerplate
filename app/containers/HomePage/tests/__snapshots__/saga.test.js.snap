// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStrings saga getStringData saga should watch for LOAD_STRINGS actions and call getStrings when one is dispatched 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getStrings saga should dispatch loadStringsError if data request fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getStrings saga should dispatch loadStringsSuccess if data request is successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/strings",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
